### DISCUSSION TITLE
getting_started/first_3d_game/01.game_setup

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#89] [User-contributed notes]
# getting_started/first_3d_game/01.game_setup

In this first part, we're going to set up the game area. Let's get started by importing the start assets and setting up the game scene. We've prepared a Godot project with the 3D models and sounds ...

https://docs.godotengine.org/en/stable/getting_started/first_3d_game/01.game_setup.html

**Comment by Wadahell** [comment]
I had to turn the blue arc for it to light floor, so watch out for that

**Reply by stefangorneanu97** [reply]
Technically, both work, bcause they're the axes of a sphere.

**Reply by ThatChapThere** [reply]
Not really. If the light is pointing along an axis rotating that axis will do nothing. It will usually be either red or blue that works, not both.

**Reply by v-kydela** [reply]
If the light is pointing straight up then there are two axes you can use to get it to point straight down

**Comment by BryceDMonaco** [comment]
It would be nice to provide assets that are already made for 4.x to skip the conversion step.

**Reply by vmorris** [reply]
I actually liked going through the conversion as I've never done it before.

**Reply by ChadAMiller** [reply]
I wouldn't mind a conversion step except I literally cannot get it to work in 4.4. It just crashes.

**Comment by firevenus** [comment]
I think it is better to let the user know how much exact number should the light rotation on XYZ should be. I had a hard time to find which should I drag and finally I made it by dragging the slide of XYZ axis, then light the Gound up.

**Reply by damienjoldersma** [reply]
I agree too, I appreciate how they did that for the Ground / CollisionShape3D / MeshInstance3D. Anyways here is what I set the Transform settings for the DirectionalLight3D too: 

Position (xyz): 0.0, 12.0, 0.0
Rotation (xyz): 270.0, 0.0, 0.0

**Reply by T-Justin96** [reply]
Fair point, but i think this is also to get Familiar with the Enginge, and with Movement and the DirectionalLight, so i kinda also appreciate the learning experience

**Comment by Erhannis** [comment]
I'm confused why the root node is a plain node rather than 3d.  Shouldn't a 3d scene start with a Node3d?  The "Create Root Node" dialog even labels it "3D Scene".

**Reply by Calinou** [reply]
A plain node doesn't have a position/rotation/scale, which avoids scenarios where you accidentally move it around. I suppose you could still use a Node3D and lock it in the editor as an alternative.

**Reply by Erhannis** [reply]
I would also argue that it's less composable: if you wanted to, say, place a room scene into a larger city scene, if the room's root node doesn't have a transform then (according to my brief tests) you can't move it around.

**Reply by ElectricZax** [reply]
Keep in mind that this is a most basic tutorial for people that have no idea what they're doing. And for you to throw them in a 'larger city scene'... let's just say it doesn't look like a basic thing to me.

**Comment by RobbiewOnline** [comment]
At this stage when I run the project it doesn't render anything like what I can see in the preview, but I suspect it's because no camera has been added to the scene yet.

It would be nice if before proceeding you could run the project and see it at runtime for confidence before moving on.

**Reply by cgmartin** [reply]
I was interested in previewing here, too, for the same reasons. I jumped ahead to [set up the camera](https://docs.godotengine.org/en/stable/getting_started/first_3d_game/03.player_movement_code.html#testing-our-player-s-movement). 

With only a single lit MeshInstance3D at this step there isn't much to visually reference. You can end up with a purely white screen in the preview, depending on position/rotation of the camera, which could be confusing to some. Having a more detailed 3d player in the later section may have been the reason for adding the camera in a later step. 

Though, getting introduced to the camera/preview as soon as possible in the tutorial seems best IMO - it is very valuable dev feedback for each step in the process. Perhaps consider moving the camera/preview instructions immediately after adding the player 3d model (and prior to setting up the player input actions).

**Comment by eddienoon** [comment]
Would I be able to make the mesh a child of the collision node itself or is that frowned upon? I am completely new to game dev and don't know many of the typical "rules" that would be used when designing a level.

**Reply by ElectricZax** [reply]
Same here (being new). I think you should stick to the tutorial. Learn the steps and bit by bit you will expand your knowledge bit by bit.

Regarding your question - it is loigcal that if it is a child - it is part of the main Collision and Mesh are on the same level, because they overlap. If they are not on the same level, but have parent/child relationship, then this could to problems.

**Comment by Catloo-Juan** [comment]
I set both the collitionshape3D and MeshInstance3D to 100 x 4 x 100 m but the MeshIntace appears way smaller than the collitionShape3D it stresses me out, also i the ground still looks grey and the player doesents interacts with it (it falls)

**Reply by SplitMilky** [reply]
Be sure you click on the BoxShape3D inside the CollisionShape3D to open the sometimes hidden size panel.   Same for the mesh click on the 3d render of the boxmesh you set for the MeshInstance3D to access its size.

**Comment by v-kydela** [comment]
Is there a way to get the mesh to match the collision shape automatically?

**Comment by v-kydela** [comment]
Can we make the mesh something other than bright white?

**Comment by web-coder-of-rpi** [comment]
How do I make the Ground node white?

**Reply by Calinou** [reply]
The ground should appear white by default already, as this is the appearance of the fallback material. (The fallback material is what's applied to every MeshInstance3D node that lacks a dedicated material.)

If the ground doesn't appear white, then check the following:

- Is the preview sun enabled (or is there a DirectionalLight3D node in the scene)?
- Is the preview environment enabled (or is there a WorldEnvironment node in the scene with a valid Environment resource)?
- Is there a rendering error in the Output panel?

**Reply by Saadorable** [reply]
All three things checked, it's still blank, both in the inspector and in the 3D view. I'm on a ryzen 5 5600 and rx 6600, so it could also probably be a driver issue? It didn't work on both Godot 4.4.1 and 4.5.1

**Reply by Calinou** [reply]
Can you take a screenshot of what it looks like?

**Reply by Soldrym** [reply]
I break on the MeshInstance3D Mesh change. It never shows up. I don't know if that's the same issue as web-code-of-rpi or Saadorable, but maybe.

When I add the BoxMesh I get an error that says:
 ERROR: Another resource is loaded from path '::rklv8' (possible cyclic resource inclusion).

If I click on BoxMesh again, output says "Set mesh" and doesn't repeat the error, but nothing shows up in inspector or the preview. Screenshot here:
<img width="2560" height="1380" alt="image" src="https://github.com/user-attachments/assets/eafae701-0067-4485-8de1-925a1aba6698" />

4.5.1 on Windows. I didn't try other versions. I am using an rx6650xt, so similar to Saadorable if that makes a difference.

Interestingly, if I nuke the MeshInstance3D and put it back, it behave the same way but I don't get a repeat error. If I delete "Ground" and recreate it, the collision shape, and the mesh instance, it does error again. I've also tried it importing the existing godot project and starting a new one (the screenshot was from a fresh project in a new empty directory). No difference there.

**Comment by solala1837** [comment]
How big should my ground be if I want to make a car driving game, to drive within a small neighborhood? Can I later change the size the of the ground later on in the game as I progress in its development?

**Reply by Calinou** [reply]
There is no set rule on how large you should make the ground. That said, if you want to make an infinite[^1] plane the car can drive on, you can use a [WorldBoundaryShape3D](https://docs.godotengine.org/en/stable/classes/class_worldboundaryshape3d.html) resource in a CollisionShape3D node.

[^1]: When using Jolt Physics, the plane is not actually infinite. See the class' description for details.

**Reply by solala1837** [reply]
Thank you!

**Reply by solala1837** [reply]
WorldBoundaryShape3D is selected as a shape of the CollisionShape3D node in the inspector, correct?May I then proceed to add a visual floor following the steps in this manual? Would my next step be to add a MeshInstance3D as a child of this CollisionShape3D? Then create a BoxMesh resource?

**Reply by Calinou** [reply]
Yes, you can use a PlaneMesh for this (it's slightly more optimal than BoxMesh). You'll also want to enable subdivision on the mesh to ensure its UVs don't suffer from precision issues. How much subdivision you need depends on the size; typically, you want to use a value between 3 and 15.

**Comment by lil-sassy** [comment]
I put together a follow-along video for this 3D project in case anyone is interested or just doesn't feel like reading.

https://www.youtube.com/watch?v=_6_76ASnCww

**Reply by alanbutidk** [reply]
hey, so i was opening this project in Godot 4.5, after that it crashed after converting the project to 4.5 from 4.2, it gave me a error that my GPU Driver are not compatible, and note that my PC doesnt have any more driver support. running with the opengl3 cmd from cmd.exe it gets stuck and doesnt even load, please maybe also make a .godot version for non-vulkan PCs!!!

**Comment by Pinaz993** [comment]
I mis-read the import instructions and ended up just importing the zip. Turns out, someone thought of that, and it worked flawlessly. Also, the assets seem to be formatted for v4.2, so no conversion is necessary.

**Reply by IndiegenesGames** [reply]
👍

**Comment by Poobslag** [comment]
The starter project should have `.godot` in the `.gitignore` file, otherwise all of Godot's transient cache data gets added to git.

**Comment by admin** [comment]
*Originally posted by [**amity**](https://github.com/amity) on 2026-06-17 23:26:43 UTC*

> Note that after moving the DirectionalLight3D up with the green arrow, you'll need to be in "Transform" or "Rotate" mode to see the red arc. If you're not seeing the arc, make sure you haven't selected "Move" in the top toolbar.

**Comment by yoshepchan490** [comment]
I added directional lights, but the Y-axis angle in the viewport changed, making it very difficult to adjust the viewpoint! What should I do?

**Comment by Just-T6202** [comment]
I've restarted this tutorial a few times now, but I am still having an issue with the player reaching the end of the CollisionShape3D and falling out of the world instead of just colliding and stopping. Using 4.7.1 stable. Is this just some step that I'm missing? Is there an additional step not listed that I need to do? Is what I think it's supposed to do not what it is actually meant to do?

**Reply by Calinou** [reply]
> but I am still having an issue with the player reaching the end of the CollisionShape3D and falling out of the world instead of just colliding and stopping.

The tutorial does not cover setting up wall StaticBodies (with CollisionShape children for each), but you can duplicate the ground and rotate it to create the walls.
